spackbot/pr_add_labels.py

```diff
@@ -0,0 +1,168 @@
+    },
+    "commands": {
+        "filename": r"^lib/spack/spack/cmd/[^/]+.py$",
+        "status": [r"^modified$", r"^modified$"],
+    },
+    "compilers": {"filename": r"^lib/spack/spack/compiler"},
+    "directives": {"filename": r"^lib/spack/spack/directives"},
+    "environments": {"filename": r"^lib/spack/spack/environment"},
+    "fetching": {"filename": r"^lib/spack/spack/(fetch|url|util/url|util/web)"},
+    "locking": {"filename": r"^lib/spack/(spack|llnl)/util/lock"},
+    "modules": {"filename": r"^lib/spack/spack/modules"},
+    "stage": {"filename": r"^lib/spack/spack/stage"},
+    "tests": {"filename": r"^lib/spack/spack/test"},
+    "utilities": {"filename": r"^lib/spack/spack/util"},
+    "versions": {"filename": r"^lib/spack/spack/version"},
+    #
+    # Documentation
+    #
+    "documentation": {"filename": r"^lib/spack/docs"},
+    #
+    # GitHub
+    #
+    "travis": {"filename": r"^\.travis"},
+    "actions": {"filename": r"^\.github/actions"},
+    "workflow": {"filename": r"^\.github/workflows"},
+    "git": {"filename": r"^\.gitignore"},
+    "flake8": {"filename": r"^\.flake8"},
+    "licenses": {"filename": r"^LICENSE"},
+    "gitlab": {"filename": r"^share/spack/gitlab"},
+    #
+    # Other
+    #
+    "defaults": {"filename": r"^etc/spack/defaults"},
+    "vendored-dependencies": {"filename": r"^lib/spack/external"},
+    "sbang": {"filename": r"sbang"},
+    "docker": {"filename": [r"[Dd]ockerfile$", r"^share/spack/docker"]},
+    "shell-support": {"filename": r"^share/spack/.*sh$"},
+}
+
+
+# compile all the regexes above, and ensure that all pattern dict values are lists
+for label, pattern_dict in label_patterns.items():
+    for attr in pattern_dict.keys():
+        patterns = pattern_dict[attr]
+        if not isinstance(patterns, list):
+            patterns = [patterns]
+        pattern_dict[attr] = [re.compile(s) for s in patterns]
+
+
+@router.register("pull_request", action="opened")
+@router.register("pull_request", action="synchronize")
+async def label_pull_requests(event, gh, *args, session, **kwargs):
+    """ Add labels to PRs based on which files were modified."""
+    pull_request = event.data["pull_request"]
+    number = event.data["number"]
+    logger.debug(f"Labeling PR #{number}...")
+
+    # Iterate over modified files and create a list of labels
+    # https://developer.github.com/v3/pulls/#list-pull-requests-files
+    labels = []
+    async for file in gh.getiter(pull_request["url"] + "/files"):
+        filename = file["filename"]
+        status = file["status"]
+        logger.debug(f"Filename: {filename}")
+        logger.debug(f"Status: {status}")
+
+        # add our own "package" attribute to the file, if it's a package
+        match = re.match(
+            r"var/spack/repos/builtin/packages/([^/]+)/package.py$", filename
+        )
+        file["package"] = match.group(1) if match else ""
+
+        # if the file's attributes match any patterns in label_patterns, add
```

Comment:
One of the ways I can tell if a section of code was written by you is if the comments are lowercase lol. According to [PEP-8](https://www.python.org/dev/peps/pep-0008/#comments):

> Comments should be complete sentences. The first word should be capitalized, unless it is an identifier that begins with a lower case letter (never alter the case of identifiers!).

Although I usually don't follow this rule for inline comments, only block comments. This is one of my many OCD things 😄

Reply:
I think I started lowercasing more based on your guidelines for command line option help years ago.  I can fix it.